assets/js/components/dashboard/SideDashboardPanel.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import MembraneLogo from "./MembraneLogo";
+import { Calendar, Package, QuestionCircle, Logout } from "react-swm-icon-pack";
+import { getFontColor } from "../../utils/styleUtils";
+import "../../../css/dashboard/sidedashboardpanel.css";
+
+type SideDashboardPanelProps = {
+  currentEvents: string;
+  setCurrentEvents: React.Dispatch<React.SetStateAction<string>>;
+};
+
+const SideDashboardPanel = ({ currentEvents, setCurrentEvents }: SideDashboardPanelProps) => {
+  const navigate = useNavigate();
+
+  const mainFontColor = getFontColor("--bg-light-color-1");
+  const bgFontColor = getFontColor("--bg-light-color-3");
+  const transitionFontColor = getFontColor("--bg-light-color-4");
+
+  const getButton = (Icon: any, text: string, onClick: () => void) => {
+    const [hoverColor, setHoverColor] = useState(bgFontColor);
+    const isActive = currentEvents == text;
+
+    return (
+      <button
+        className="DashboardPanelButton"
+        onClick={onClick}
+        // a very inconvinient way to add hover effect to the icon
+        // as it seems to be impossible in css
+        onMouseEnter={() => setHoverColor(transitionFontColor)}
+        onMouseLeave={() => setHoverColor(bgFontColor)}
```

Comment:
Did you ask someone more proficient in frontend what is a better way to do this?

Reply:
Fixed.